### ISSUE TITLE
Nick Formatting

### ISSUE THREAD

**Issue by mercurialmusic** [Zrips/CMI#252]
Description of issue or feature request:
#236 hasn't quite been fixed for us yet. While the nicks themselves now seem to work, CMI won't pull in the Prefixes (from Vault/LuckPerms) as defined by our desired Nickname Format: '{prefix}{nickName}'. The opposite was true in 7.6.5.2 where it was only pulling the prefixes and not the nicks.

If the tablist is any indication, what I think would solve it for us is if that nickname formatting config option accepted your other placeholders. We have our tablist set to: PlayerName: '%cmi_user_prefix%%cmi_user_display_name%' and it works exactly how we want it to. Then of course those are probably completely different systems and adding placeholders would kill everything. Just a thought.

Cmi Version (using`/cmi version`): 7.6.5.4
Server Type (Spigot/Paperspigot/etc): Spigot
Server Version (using `/ver`): The latest one =p

**Comment by MCTyler** [comment]
Is there another Chat Plugin involved cause mine works fine?

**Comment by mercurialmusic** [comment]
We use [VentureChat](https://www.spigotmc.org/resources/venturechat.771/) and [NametagEdit](https://www.spigotmc.org/resources/nametagedit.3836/). VentureChat does have a nick feature, although we don't use it, and NTE does have a prefix feature, but that modifies the actual tag above someone's head -- and anyway we have it set to the same prefix as our permissions groups so if that is interfering it should work out the same. We have CMI's ModifyChatFormat: false, so those group format's you're mentioning (while ours are the same) should not apply. 

I'm more concerned with the NickName feature quite a bit farther down the config. This was working in the same way the same feature did on Essentials, which we had also used with the same setup without issue. That, I think, is how most people will expect it to work here as well. Other than the issue with nicks taking a while to save, it was working just fine for us until 3.6.5.2, which makes me think it's not an issue with any other plugin.

Currently, as I said, the nicks themselves do persist, but it no longer pulls the prefix from the NickName format (not the chat format). Our workaround at the moment is to set the nicks again every time someone logs in -- this pulls in the prefix as it should, but then does not persist at all after a relog no matter how much time elapsed.

**Comment by LogGits** [comment]
Wait so the issue is nicknames not persisting after restarts/reloads? Out of curiosity are you using essentials to change the nicks? And are you using cmi vault?

**Comment by mercurialmusic** [comment]
Correct. The **username** changes now stick, but the formatting to bring the prefix into the display name does not and resets with a relog. Even for players who do not have a nick on our sever I have to run /cmi nick LogGits LogGits just so their prefix will show up in game. When they relog that goes away. Same deal for those who do have a nick: I'll run /cmi nick merc mercurialmusic and my prefix will show up with the nick, but if I relog it's just back to "merc" without the prefix.

Yes, we exclusively use CMI for nicks, and have aliased all the /nick commands to point to CMI, but with the current behavior we've been using /cmi nick just to be extra sure. And yep, we're using the CMI Vault, version 1.6.9.

I did try using the placeholders in VentureChat, but they all just printed as the placeholder itself, even after restarts. That, I think, is a limitation of VentureChat since it has it's own "{nickname}" placeholder that's used, but as I mentioned that has worked without issue until CMI 3.6.5.2. I've been looking for an excuse to switch to a new chat plugin and have been dragging my feet about it in sort of a "the devil you know" way.

When I said "other than the issue with nicks taking a while to save" I figured this was a known issue since I've seen several people commenting on it on the Spigot board and occasionally here. Simply that, though: ever since we started using CMI nicks they would not seem to save to the database (SQLite) for quite a while. We would have to catch the player right when they logged in to change their nick, then hoped it would have enough time to save to the database before they logged out. I was never able to distinguish a pattern in how long it taked to save, but typically the player would have to stay logged in for an hour or more before the nick would persist after a relog. We literally built a room off our staff lounge full of command blocks housing people's nick commands so we could quickly tp there when they logged in and click the button with their nick so we didn't a) have to remember everyone's nick and b) type out the entire command.  The issue now is similar and we've been using the same system.

**Comment by mercurialmusic** [comment]
Just wanted to say this is working perfectly now. Thank you, thank you, thank you!
